src/Unirest/Request.php

```diff
@@ -496,34 +514,47 @@ public static function buildHTTPCurlQuery($data, $parent = false)
      * @param array $headers additional headers to send
      * @param string $username Authentication username (deprecated)
      * @param string $password Authentication password (deprecated)
-     * @throws \Unirest\Exception if a cURL error occurs
      * @return Response
+     * @throws \Unirest\Exception if a cURL error occurs
      */
     public static function send($method, $url, $body = null, $headers = array(), $username = null, $password = null)
     {
-        self::$handle = curl_init();
+        if (self::$handle == null) {
+            self::$handle = curl_init();
+        }
+
+        else {
+            self::$prevCallSuccessfulConnects = curl_getinfo(self::$handle, CURLINFO_NUM_CONNECTS);
+
+            curl_reset(self::$handle);
+        }
 
         if ($method !== Method::GET) {
-			if ($method === Method::POST) {
-				curl_setopt(self::$handle, CURLOPT_POST, true);
-			} else {
-                 if ($method === Method::HEAD) {
+            if ($method === Method::POST) {
+                curl_setopt(self::$handle, CURLOPT_POST, true);
+            }
+            else {
+                if ($method === Method::HEAD) {
                     curl_setopt(self::$handle, CURLOPT_NOBODY, true);
-                 }
-				curl_setopt(self::$handle, CURLOPT_CUSTOMREQUEST, $method);
-			}
+                }
+                curl_setopt(self::$handle, CURLOPT_CUSTOMREQUEST, $method);
+            }
 
             curl_setopt(self::$handle, CURLOPT_POSTFIELDS, $body);
-        } elseif (is_array($body)) {
+        }
+        elseif (is_array($body)) {
             if (strpos($url, '?') !== false) {
                 $url .= '&';
-            } else {
+            }
+            else {
                 $url .= '?';
             }
 
             $url .= urldecode(http_build_query(self::buildHTTPCurlQuery($body)));
         }
 
+        array_push(self::$defaultHeaders, "Keep-Alive: timeout=" . self::$connectionTimeout);
```

Comment:
This is causing a big issue in the headers of an endpoint call, each time a call is made, this line inserts a new value in the default headers, causing the array size to grow over the time